awx/main/tasks.py

```diff
@@ -1138,6 +1139,23 @@ def event_handler(self, event_data):
             else:
                 event_data['host_name'] = ''
                 event_data['host_id'] = ''
+
+        if isinstance(self, RunProjectUpdate):
+            # it's common for Ansible's SCM modules to print
+            # error messages on failure that contain the plaintext
+            # basic auth credentials (username + password)
+            # it's also common for the nested event data itself (['res']['...'])
+            # to contain unredacted text on failure
+            # this is a _little_ expensive to filter
+            # with regex, but project updates don't have many events,
+            # so it *should* have a negligible performance impact
+            try:
+                event_data_json = json.dumps(event_data)
```

Comment:
Historically, we've filtered this data on the *API* side, but I think we should consider doing it here, too, because right now we *ship* this raw data to external loggers (where it may be unredacted).